src/Docker/VariableResolver.php

```diff
@@ -0,0 +1,67 @@
+<?php
+
+namespace Keboola\DockerBundle\Docker;
+
+use Keboola\DockerBundle\Docker\Configuration\Variables;
+use Keboola\DockerBundle\Docker\Configuration\VariableValues;
+use Keboola\DockerBundle\Exception\UserException;
+use Keboola\StorageApi\Client;
+use Keboola\StorageApi\Components;
+use Mustache_Engine;
+
+class VariableResolver
+{
+    /**
+     * @var Client
+     */
+    private $client;
+
+    const KEBOOLA_VARIABLES = 'keboola.variables';
+    /**
+     * @var Mustache_Engine
+     */
+    private $moustache;
+
+    public function __construct(Client $client)
+    {
+        $this->client = $client;
+        $this->moustache = new Mustache_Engine();
+    }
+
+    public function resolveVariables(array $jobDefinitions, $varaiableValuesId, $variableValuesData)
+    {
+        if ($varaiableValuesId && $variableValuesData) {
+            throw new UserException('Only one of variableValuesData and variableValuesId can be entered');
+        }
+        /** @var JobDefinition $jobDefinition */
+        $newJobDefinitions = [];
+        foreach ($jobDefinitions as $jobDefinition) {
+            if ($jobDefinition->getConfiguration()['variables_id']) {
```

Comment:
to $jobDefinition->getConfiguration() by som vytiahol do premennej, lebo sa to dalej vola niekolkokrat a ten zapis je zbytocne dlhy.